src/Quasar/Advanced/Types.purs

```diff
@@ -146,27 +106,24 @@ parseDir pt =
   <#> (Pt.rootDir </> _)
   # maybe (Left "Incorrect resource") pure
 
+
 type ActionR =
   { operation ∷ Operation
   , resource ∷ Resource
   , accessType ∷ AccessType
   }
 
-newtype Action = Action ActionR
+newtype Action = Action
```

Comment:
Well, kinda... `ActionR` is used in `QuasarAF`, and also in `runAction`. I changed the newtype to avoid `ActionR` so that we can use deriving for the instances. This is why I preferred not to use newtypes at all and wrote the codecs without using instances for the records in the other part.